adds a new feature to cooking: food quality

timothyteakettle opened the pull request:
## About The Pull Request
the main thing: food quality
all food items have quality, ranging from 0-100 (default 50, causing no real change to most foods)
food quality is based off potency (0 potency, 0 quality, 100 potency, 100 quality)
everything that is crafted calculates quality off of the ingredients
you like foods with high quality and dislike foods of a lower quality

the other thing: frying
condensed cooking oil now has a 5% chance to vomit
reagents from bad fried items get transferred to you when you eat them, and are produced only at that moment (and thus do not exist in the item itself, so you can't grind them to get the chemicals)

## Why It's Good For The Game
encourages cooks to give you high potency ingredients, moodlets for good food can be achieved without it having to be something your species likes (as long as it isn't something you dislike)

more fun for cooks as they can now aim to be a better cook by producing high quality food, and adds more meaning to upgrading the cooks machines other than 'you have more food'

more buffs could probably be given to high quality food later on, such as reducing the amount of healing nutriment does, and making eating high quality food increase the effect of nutriment

## Changelog
:cl:
add: food quality is now a property of all food, food gets a description for if it is high/low quality (above or below 70)
add: splashing 1u or more of secret sauce on food sets its quality to 100
tweak: kitchen machines have larger effect on food quality as they are upgraded
tweak: you only get a good moodlet for eating food that is not disliked and either 70+ quality, or liked by your species with 50+ quality, disgust reduction scales with quality
tweak: condensed cooking oil only does a 5% vomit chance
tweak: bad frying reagents are placed into the consumer when eating a fried inedible item instead of residing in the food
tweak: butchered animals give meat of higher quality if butchered in the kitchen and/or are butchered by better tools (including the family knife)
del: secret sauce doesn't have a good food quality now, and instead of awarding moodlets like it used to, it just buffs food quality (which in turn makes food give moodlets)
/:cl:

timothyteakettle committed:
i too, like food quality.

timothyteakettle committed:
more food quality

timothyteakettle committed:
woops

timothyteakettle committed:
more quality

timothyteakettle committed:
more quality

timothyteakettle committed:
people are more picky about food

timothyteakettle committed:
fried garbage is now better

timothyteakettle committed:
more quality

timothyteakettle committed:
more quality

timothyteakettle committed:
what if butchering didnt suck!

timothyteakettle committed:
makes descriptions work

timothyteakettle committed:
make secret sauce work on customizables

timothyteakettle committed:
imagine making it compile

Linzolle commented on code/modules/food_and_drinks/food.dm:
```diff
@@ -32,13 +36,13 @@
 					H.adjust_disgust(11 + 15 * fraction)
 					SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "gross_food", /datum/mood_event/gross_food)
-				else if(foodtype & H.dna.species.liked_food)
+				else if((foodtype & H.dna.species.liked_food & food_quality >= 50) || food_quality >= 70) //you like food of high quality, and food of regular quality you have a preference for
```
```suggestion
				else if(((foodtype & H.dna.species.liked_food) && food_quality >= 50) || food_quality >= 70) //you like food of high quality, and food of regular quality you have a preference for
```
you put a bitwise operator instead of a boolean

timothyteakettle committed:
swap & for &&

woops

Co-Authored-By: Lin <linzolle@gmail.com>

timothyteakettle committed:
secret sauce doesn't give a second moodlet because that's dumb

Ghommie commented on code/modules/food_and_drinks/food/snacks.dm:
```diff
@@ -291,11 +297,15 @@ All foods are distributed among various categories. Use common sense.
 	if(cooked_type)
 		result = new cooked_type(T)
+		//if the result is food, set its food quality to the original food item's quality
+		if(istype(result, /obj/item/reagent_containers/food))
```
You should add an `isfood` helper define (see __DEFINES/is_helpers.dm) at this point.

timothyteakettle replied:
done

Ghommie commented on code/modules/food_and_drinks/food.dm:
```diff
@@ -16,13 +16,17 @@
 		pixel_y = rand(-5, 5)
 
+/obj/item/reagent_containers/food/proc/adjust_food_quality(new_quality)
+	food_quality = clamp(new_quality,0,100)
```
This could as well be a defined macro unless subtypes with other effects based on food quality are planned.
Also, until we finally butcher 512 compatibily (which will be soon):
```suggestion
	food_quality = CLAMP(new_quality,0,100)
```

Ghommie commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -772,11 +770,20 @@
+		if(istype(O, /obj/item/reagent_containers/food/snacks/customizable))
+			var/obj/item/reagent_containers/food/snacks/customizable/splashed_custom_food = O
+			splashed_custom_food.total_quality += 10000
+r
```
I doubt that's a proper whitespace line, my dude. :thinking:

timothyteakettle replied:
woops

timothyteakettle committed:
change clamp to CLAMP

requested change, and i don't plan on subtypes with different effects based on food quality

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

timothyteakettle committed:
remove a random 'r'

timothyteakettle committed:
implements isfood helper

silicons commented on code/modules/food_and_drinks/food/snacks.dm:
```diff
@@ -291,11 +297,15 @@ All foods are distributed among various categories. Use common sense.
 				S.reagents.add_reagent(r_id, amount)
 
-/obj/item/reagent_containers/food/snacks/microwave_act(obj/machinery/microwave/M)
+/obj/item/reagent_containers/food/snacks/microwave_act(obj/machinery/microwave/M, var/quality_increase)
```
take out the var/, it's implied in proc header/whatever it's called

timothyteakettle replied:
i'm just gonna remove it as a parameter because it's defined in obj/machinery/microwave meaning it's not needed to be passed to microwave_act

silicons commented on code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:
```diff
@@ -104,6 +104,12 @@ God bless America.
+				var/obj/item/reagent_containers/food/original_food = I
+				frying.adjust_food_quality(original_food.food_quality) //food quality remains unchanged until degree of frying is calculated
+			else
+				frying.food_quality = 10 //inedible fried item has low quality
```
er is this supposed to not use adjust_food_quality for some reason because you're using that everywhere else

timothyteakettle replied:
all changes to it should be done by adjust food quality ill change that

silicons commented on code/modules/food_and_drinks/kitchen_machinery/gibber.dm:
```diff
@@ -23,8 +24,10 @@
 /obj/machinery/gibber/RefreshParts()
 	gibtime = 40
 	meat_produced = 0
+	meat_quality = 35 // unupgraded this means quality is 50, and max upgraded it is 95
```
might be a good idea to make sure this is clamped between 0 and 100

timothyteakettle replied:
as long as the final product (gibber meat) has its quality set using adjust food quality then it should already be clamped between 0 and 100 (edit: and I checked, it does have its quality set as such)

timothyteakettle committed:
small refactor to some stuff

quality_increase no longer passed to microwave_act because it's defined in obj/machinery/microwave

deepfryer code uses adjust_food_quality in a place where it didn't before

timothyteakettle committed:
Merge branch 'master' into food-quality

timothyteakettle committed:
Merge branch 'master' into food-quality

Ghommie reviewed (approved):
I'm gonna regret this should I port the edible component.
